network: Add nmstate examples (#536)
coreos/fedora-coreos-docs: modules/ROOT/pages/sysconfig-network-configuration.adoc

WARNING: If you need networking to grab your Ignition config and your environment requires more complex networking than the default of DHCP to grab the Ignition config, then you'll need to use another method other than Ignition to configure the network.

> There are to main methods for configuring networking with ignition 

Reviewer: ```suggestion
There are two main methods for configuring networking by delivering text configuration files via Ignition:
```

Also Let's add a blank line after this line so the bullets get broken out properly by antora/asciidoctor

> - By creating NetworkManager configuration files described in an Ignition config. These are https://networkmanager.dev/docs/api/latest/nm-settings-keyfile.html[NetworkManager keyfiles] that are written to `/etc/NetworkManager/system-connections/` that tell NetworkManager what to do.

Reviewer: I think we can shorten this now:

```suggestion
- By creating https://networkmanager.dev/docs/api/latest/nm-settings-keyfile.html[NetworkManager keyfiles] that are written to `/etc/NetworkManager/system-connections/` that tell NetworkManager what to do.
```

> - By creating a nmstate desire state YAML(postfix .yml) files at /etc/nmstate following https://nmstate.io[NMstate syntax]. The applied network state file will be renamed with postfix .applied to prevent repeated applied on next run.

Reviewer: I assume everywhere in this file we should be consisten with the capitalization of `NMstate`? Also a few wording suggestsion:

```suggestion
- By creating NMstate desired state YAML files with the `.yml` suffix under `/etc/nmstate/` following the https://nmstate.io[NMstate syntax]. The state files will be applied and then renamed with a `.applied` suffix to prevent being processed again.
```

Any configuration provided via Ignition will be considered at a higher priority than any other method of configuring the Network for a Fedora CoreOS instance. If you specify Networking configuration via Ignition, try not to use other mechanisms to configure the network.

An example https://docs.fedoraproject.org/en-US/fedora-coreos/producing-ign/[Butane] config for the same static networking example that we showed above is:

> NetworkManager

Reviewer: Adding a `.` on the front will make the formatting look nice in antora:

```suggestion
.NetworkManager keyfile
```

Also suggested adding `keyfile` on the end there.

I would apply this to all other instances in the file.

          type=ethernet
          interface-name=ens2
          [ipv4]
          address1=10.10.10.10/24
          dns=8.8.8.8;
          dns-search=
          may-fail=false
          method=manual
----

> NMstate

Reviewer: The `.` on the front makes the formatting nicer. Also suggest adding YAML:

```suggestion
.NMstate YAML
```

I would apply this to all other instances in the file.

[source, yaml]
----
variant: fcos
version: 1.4.0
storage:
  files:
    - path: /etc/nmstate/ens2.yml
      mode: 0600
      contents:
        inline: |
          hostname:
            config: myhostname
          dns-resolver:
            config:
              search: []
              server:
              - 8.8.8.8
          interfaces:
          - name: ens2
            type: ethernet
            wait-ip: ipv4
            ipv4:
              enable: true
              address:
              - ip: 10.10.10.10

>                 prefix-length: 24

Reviewer: the later examples include a `routes:` entry in the nmstate yml file. I'm not sure I understand why those include it and this one doesn't?

----
== Host Network Configuration Examples

> In this section we'll go through common examples of setting up different types of networking devices using dracut kernel arguments, NetworkManager keyfiles and nmstate via Ignition/Butane.

Reviewer: ```suggestion
In this section we'll go through common examples of setting up different types of networking devices using dracut kernel arguments, NetworkManager keyfiles and NMstate via Ignition/Butane.
```

ip=10.10.10.10::10.10.10.1:255.255.255.0:myhostname:ens2:none:8.8.8.8
----

> ==== Butane NetworkManager config

Reviewer: To make it a little more clear:

```suggestion
==== Butane NetworkManager keyfile config
```

If we make this change then apply everywhere else too

          type=ethernet
          interface-name=ens2
          [ipv4]
          address1=10.10.10.10/24
          dhcp-hostname=myhostname
          dns=8.8.8.8;
          dns-search=
          may-fail=false
          method=manual
----

> ==== Butane Nmstate config

Reviewer: ```suggestion
==== Butane NMstate YAML config
```

If we make this change then apply everywhere else too
